src/server/object_services/map_deleter.js

```diff
@@ -43,6 +44,17 @@ async function delete_chunks_if_unreferenced(chunk_ids) {
     }
 }
 
+/**
+ * @param  {nb.ObjectMD} object
+*/
+async function delete_object_if_no_parts(object) {
+    if (!object) return;
+    dbg.log1('delete_object_if_no_parts: object_ids', object);
+    if (!MDStore.instance().has_any_parts_for_object(object)) {
```

Comment:
missing await, but I would not add it inside an `if` so I suggest:
```suggestion
    const has_parts = await MDStore.instance().has_any_parts_for_object(object);
    if (!has_parts) {
```